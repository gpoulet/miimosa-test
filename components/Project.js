import { string, number } from "prop-types"
import Link from "next/link"

function Project({ id, title, collected, target, status }) {
  return (
    <div>
      <p>
      id: {id}<br/>
      title: {title}<br/>
      collected: {collected}<br/>
      target: {target}<br/>
      status: {status}<br/>
      <Link href={"/details/"+id}>
        <a>DÃ©couvrir</a>
      </Link>
      </p>
    </div>
  )
}

Project.propTypes = {
  id: string.isRequired,
  title: string.isRequired,
  collected: number.isRequired,
  target: number.isRequired,
  status: string.isRequired,
};

export default Project;

//
// "collected": 15000,
//   "description": "",
//   "id": "c6d6f6f9-07d8-4540-87a7-3b8ff8869f55",
//   "image_url": "",
//   "status": "success",
//   "target": 15000,
//   "title": "Bar smoothie Bio",
//   "user": {
//     "first_name": "Lancelot",
//     "last_name": "Little Finger"
// }