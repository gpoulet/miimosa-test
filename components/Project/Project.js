import { string, number } from "prop-types"
import Money from '../Money'
import CardLink from '../CardLink'
import Title from '../Title'
import Photo from '../Photo'
import Card from "@material-ui/core/Card"
import CardContent from '@material-ui/core/CardContent'
import CardActions from "@material-ui/core/CardActions"

function Project({ id, title, collected, target, status }) {
  return (
    <Card className="projectCard">
      <Photo id={id} href={"/details/" + id} label={title} />
      <CardContent>
        <Title title={title}/>
        <Money collected={collected} target={target} status={status}/>
      </CardContent>
      <CardActions>
        <CardLink href={"/details/" + id} label="DÃ©couvrir le projet"/>
      </CardActions>
    </Card>
  )
}

Project.propTypes = {
  id: string.isRequired,
  title: string.isRequired,
  collected: number.isRequired,
  target: number.isRequired,
  status: string.isRequired,
  description: string.isRequired,
  imageUrl: string
};

export default Project;