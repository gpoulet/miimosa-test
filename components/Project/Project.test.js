import React from 'react';
import renderer from 'react-test-renderer';
import Project from './Project';

it('Project in progress renders correctly', () => {
  const tree = renderer
    .create(<Project
      id="test"
      collected={2000}
      target={5000}
      status="in_progress"
      description="lorem ipsum"
      title="La ferme magique"
      user={{first_name: "Albus", last_name: "Dumbledore"}}
      imageUrl="/fermemagique.jpg"
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Project failed renders correctly', () => {
  const tree = renderer
    .create(<Project
      id="test"
      collected={34}
      target={5000}
      status="failed"
      description="lorem ipsum bla"
      title="La ferme sans machine"
      user={{first_name: "Abelforth", last_name: "Dumbledore"}}
      imageUrl="/fermemagique.jpg"
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Project success renders correctly', () => {
  const tree = renderer
    .create(<Project
      id="test"
      collected={10000}
      target={8000}
      status="success"
      description="lorem ipsum bla bla"
      title="La ferme sans Ã¢me"
      user={{first_name: "Tom", last_name: "Jedusor"}}
      imageUrl="/fermesansame.png"
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});