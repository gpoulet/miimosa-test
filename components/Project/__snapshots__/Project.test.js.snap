// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project failed renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root projectCard MuiPaper-elevation1 MuiPaper-rounded"
>
  <a
    href="/details/test"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="MuiCardMedia-root"
      style={
        Object {
          "backgroundImage": "url(\\"/images/test.png\\")",
          "height": 0,
          "paddingTop": "40%",
        }
      }
      title="La ferme sans machine"
    />
  </a>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      La ferme sans machine
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      34€ collectés sur 5000€ (0%)
    </p>
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={1}
      className="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-99.32%)",
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <a>
          Découvrir le projet
        </a>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Project in progress renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root projectCard MuiPaper-elevation1 MuiPaper-rounded"
>
  <a
    href="/details/test"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="MuiCardMedia-root"
      style={
        Object {
          "backgroundImage": "url(\\"/images/test.png\\")",
          "height": 0,
          "paddingTop": "40%",
        }
      }
      title="La ferme magique"
    />
  </a>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      La ferme magique
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      2000€ collectés sur 5000€ (40%)
    </p>
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={40}
      className="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-60%)",
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <a>
          Découvrir le projet
        </a>
      </span>
    </button>
  </div>
</div>
`;

exports[`Project success renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root projectCard MuiPaper-elevation1 MuiPaper-rounded"
>
  <a
    href="/details/test"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="MuiCardMedia-root"
      style={
        Object {
          "backgroundImage": "url(\\"/images/test.png\\")",
          "height": 0,
          "paddingTop": "40%",
        }
      }
      title="La ferme sans âme"
    />
  </a>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      La ferme sans âme
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      10000€ collectés sur 8000€ (125%)
    </p>
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={100}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(0%)",
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <a>
          Découvrir le projet
        </a>
      </span>
    </button>
  </div>
</div>
`;
